Userland/Libraries/LibWeb/HTML/HTMLCanvasElement.cpp

```diff
@@ -285,8 +289,12 @@ String HTMLCanvasElement::to_data_url(StringView type, Optional<double> quality)
 }
 
 // https://html.spec.whatwg.org/multipage/canvas.html#dom-canvas-toblob
-WebIDL::ExceptionOr<void> HTMLCanvasElement::to_blob(JS::NonnullGCPtr<WebIDL::CallbackType> callback, StringView type, Optional<double> quality)
+WebIDL::ExceptionOr<void> HTMLCanvasElement::to_blob(JS::NonnullGCPtr<WebIDL::CallbackType> callback, StringView type, JS::MarkedVector<JS::Value> arguments)
 {
+    Optional<double> quality;
+    if (arguments.size() >= 1 && arguments[0].is_number())
+        quality = arguments[0].as_double();
+
```

Comment:
This step should be performed inside serialize_bitmap.

The behavior is specified in this section:

https://html.spec.whatwg.org/multipage/canvas.html#a-serialisation-of-the-bitmap-as-a-file

So the exact text should be pasted above the normalization:

> If type is an image format that supports variable quality (such as "[image/jpeg](https://html.spec.whatwg.org/multipage/indices.html#image/jpeg)"), quality is given, and type is not "[image/png](https://html.spec.whatwg.org/multipage/indices.html#image/png)", then, if quality [is a Number](https://tc39.es/ecma262/#sec-ecmascript-language-types-number-type) in the range 0.0 to 1.0 inclusive, the user agent must treat quality as the desired quality level. Otherwise, the user agent must use its default quality value, as if the quality argument had not been given.

Reply:
Hmm, I'm not sure I understand the second part. As long as it's a Number between 0.0 and 1.0, it's valid.  Otherwise, treat it as if no quality argument was given. As far as I can tell that's what's happening, with the checking of the range being done in serialize_bitmap. Am I missing something?

Reply:
Combined with Tim's comment about the signature being incorrect, My comment boils down to "do all the validation on the argument in `serialize_bitmap`